SubscriptionSidebar: improvements and default changes

BentiGorlich opened the pull request:
- show subs by default
- add icons in the header of the panel to pop it in and out and, if popped out, to change the side the panel is on
- add a setting to make the inline sub panel larger (the former size)
- set the default `max-height` of the inline panel to `20em` which is half the size
- make the popped out panel smaller (half of the sidebar and a 6th of the main content -> double the ratio on everything except the sub sidebar
- hide the placeholder div in popped out state so no space is wasted
- add the sort values to the `VALUES` array in `ThemeSettingsController`

Fix #154

A lot of this is up for debate. I just implemented it in a way that should satisfy @TheVillageGuy and the way I'd want it myself.

# Images

1. Freshly logged in: 
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/7de48715-8f19-4f85-bebf-7ef3b8b7c0a6)

2. After clicking the left arrow:
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/3369b206-be4a-4426-9f32-0f7517d073e5)

3.After clicking the right arrow:
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/23ae89dc-a016-48dd-a9bf-71d2434b89dd)

4. After clicking the right arrow again:
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/49d855ba-6182-4697-aaf5-9dc5db0f07a8)

4. After clicking the left arrow:
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/585d09ee-51b7-4c24-a0c9-03f75dc2d091)

6. After changing the "Subscriptions panel large" setting to true:
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/466362eb-f6d3-4d50-84f9-fdd765ecf4a7)

7. After changing the "Page width" to "Auto":
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/3a11a5b6-3a3b-46d7-982e-8fa58b07fe77)

8. After changing the "Page width" to "Max":
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/fef44cb3-0a89-4dd9-a544-021911b61ed5)

9. After clicking the left arrow again:
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/7cad6632-8dec-435d-bfed-73141947a926)

10. After clicking the right arrow:
![grafik](https://github.com/MbinOrg/mbin/assets/25664458/2894ffdc-8326-4ffa-9b14-45333285d7af)

melroy89 commented:
You could also just trim it at the end, like a max. number of characters are allowed and if that exceeds you just limit the chars then just add `...` at the end. Something like that. 

If we enable this by default.. I really really believe that this needs to work together with a bigger width in general therefore see related PR: https://github.com/MbinOrg/mbin/pull/121

BentiGorlich commented:
> You could also just trim it at the end, like a max. number of characters are allowed and if that exceeds you just limit the chars then just add `...` at the end. Something like that.

I considered that, but
1. I didn't manage to implement it. What CSS would that be or would you do it in the backend?
2. I think it looks just fine even with the longes name, but maybe a lot of people disagree with that :D

TheVillageGuy commented:
The button system is spot on.

TheVillageGuy commented:
The bar seems narrower than the regular sidebar? I liked it symmetrical, but this isn't bad either. However, many magazine names will be too wide and the way it works now, for instance with "Langer Magazine Name", does not look good. Instead perhaps we could use a lighter font, end the line with ... and with a hover show the full name,

melroy89 commented:
> The bar seems narrower than the regular sidebar? I liked it symmetrical, but this isn't bad either. However, many magazine names will be too wide and the way it works now, for instance with "Langer Magazine Name", does not look good. Instead perhaps we could use a lighter font, end the line with ... and with a hover show the full name,

True.. I actually requested to have the width smaller, in order to have the main body the same size, but I was aware this will create a asymmetrical look & feel as a down-side. Unless there is a better solution.

melroy89 commented:
> > You could also just trim it at the end, like a max. number of characters are allowed and if that exceeds you just limit the chars then just add `...` at the end. Something like that.
> 
> I considered that, but
> 
>     1. I didn't manage to implement it. What CSS would that be or would you do it in the backend?
> 
>     2. I think it looks just fine even with the longes name, but maybe a lot of people disagree with that :D

https://stackoverflow.com/questions/11426275/how-can-i-show-dots-in-a-span-with-hidden-overflow

e-five256 commented on templates/layout/sidebar_subscriptions.html.twig:
```diff
@@ -1,6 +1,43 @@
+            {% endif %}
+        </span>
+    </h3>
+    <div class="subscription-list {{ app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_GENERAL_SHOW_SUBSCRIPTIONS_LARGE')) is same as 'true' ? 'lg' : '' }}">
```
I think if you add the class `meta` here, you'll get a lot of the styling to match the other widgets. For example the `.75` opacity and the text color

before:
![image](https://github.com/MbinOrg/mbin/assets/146029455/9932625d-9448-4d40-9adb-fd27370e7f88)

after:
![image](https://github.com/MbinOrg/mbin/assets/146029455/b25740f4-433a-4cb3-a470-c9799de99dfb)

e-five256 commented:
on https://github.com/MbinOrg/mbin/pull/161/files#diff-238b2d267292fab3559e0698428914b62472452bb0a82a78128257dafe967505R46 do you have thoughts on adding

```
class="image-inline"
  width="30" height="30"
  style="max-width: 30px; max-height: 30px;"
```

to match https://github.com/MbinOrg/mbin/blob/main/templates/components/magazine_inline.html.twig#L5-L7

---

I would like to add at least `image-inline` as that protects against overflow when images fail to load, but not sure about opinions on the size of the icon

BentiGorlich commented:
The image overflow problem should be fixed now by setting max-width and -height

BentiGorlich commented:
I also rebased the branch and it looks fine with the code from #121

melroy89 commented:
Are the screenshots still accurate in your description?

e-five256 commented:
I, personally, am not sure it should be included on the magazine, entry, and non-top level microblog pages. (basically everywhere random doesn't appear in the sidebar). if you're already at a magazine view or in a thread I think you don't really need to see all your subscriptions, it sort of makes the page too busy / confusing UX flow

BentiGorlich commented:
Imo it should appear. I want to be able to switch between my subbed magazines easily and to remove the panel from everything but the main page is not ideal...

BentiGorlich commented:
> Are the screenshots still accurate in your description?

I'll update them.

e-five256 commented on assets/styles/components/_sidebar-subscriptions.scss:
```diff
@@ -59,9 +75,16 @@
+        max-width: 1.5rem;
         object-fit: scale-down;
         border-radius: 50%;
         display: inline;
```
I think this is overriding the `image-inline` class, so the alt text is overflowing (`image-inline` sets a `inline-block` display)

so it's doing this when the images break
![image](https://github.com/MbinOrg/mbin/assets/146029455/4741d885-004a-4bbb-a378-9eb6fe36fd3a)

rather than
![image](https://github.com/MbinOrg/mbin/assets/146029455/12d861f7-900a-45f0-8116-f09680538d10)

BentiGorlich replied:
you're right, I'll remove that :+1:

BentiGorlich commented:
> > Are the screenshots still accurate in your description?
> 
> I'll update them.

done

e-five256 commented:
The styling when separate looks a bit different to me, which I'm not sure if it's because of the meta thing I mentioned earlier or not, still looking into it, but this is what I see

![image](https://github.com/MbinOrg/mbin/assets/146029455/b3bef2a9-3829-4c24-b6ba-94eb6083288b)

BentiGorlich commented on templates/layout/_options_appearance.html.twig:
```diff
@@ -17,16 +17,16 @@
                 {{ component('settings_row_enum', {label: 'subscription_sort'|trans, settingsKey: 'KBIN_GENERAL_SHOW_SUBSCRIPTIONS_SORT', values: [ {name: 'alphabetically'|trans , value: 'ALPHABETICALLY'}, {name: 'last_active'|trans , value: 'LAST_ACTIVE' } ], defaultValue: 'LAST_ACTIVE'}) }}
-            {% endif %}
-            {% if app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_GENERAL_SHOW_SUBSCRIPTIONS')) is same as 'true' %}
                 {{ component('settings_row_switch', {label: 'subscriptions_in_own_sidebar'|trans, settingsKey: 'KBIN_GENERAL_SHOW_SUBSCRIPTIONS_IN_SEPARATE'}) }}
```
I thought it would be to make the settings appear to more users. But I think you're right and they could be removed.

e-five256 replied:
I think this is my only outstanding comment: I would say get rid of those two options in the sidebar options as they've moved to the widget itself

BentiGorlich replied:
I'll do that then

e-five256 replied:
wait which one? :stuck_out_tongue_closed_eyes: if the latter and hiding it on mobile, I did that with the page width toggle in this same file, you could potentially rename that from `width-setting` to something generic like `hide-on-mobile` and use it for all 3 toggles. if the former can ignore this comment

BentiGorlich replied:
I just hid the two options, but now I realize, that you have no way of making the sidebar appear at the top on mobile, because you can't make the sidebar separate from the rest :D

BentiGorlich replied:
If I could easily just always put the sidebar on top if the user is on mobile that would be very nice... But nobody told me an easy way to recognize mobile user agents :sweat_smile:

BentiGorlich commented:
But I really don't know why the icons appear so weird...

BentiGorlich commented:
> > The styling when separate looks a bit different to me, which I'm not sure if it's because of the meta thing I mentioned earlier or not, still looking into it, but this is what I see
> > ![image](https://user-images.githubusercontent.com/146029455/279127446-b3bef2a9-3829-4c24-b6ba-94eb6083288b.png)
> > not sure why the dotted line is appearing above all the content rather than at the bottom, and where the icons are
> 
> That's still an error. The CSS that adds that was more specific for the popped out panel than mine which reverts it.

fixed

e-five256 reviewed (dismissed):
looks good to me

e-five256 commented on translations/messages.en.yaml:
```diff
@@ -794,6 +794,11 @@ subscription_sort: Subscription sort
 alphabetically: Alphabetically
 subscriptions_in_own_sidebar: Subscriptions separate
 sidebars_same_side: Sidebars on the same side
```
I think this and the above translation can be cleaned up but doesn't need to be part of this PR, maybe we can wait and see if people disagreed with me and want it back :laughing:

BentiGorlich replied:
you can always change translations in weblate :D

e-five256 replied:
can `en` translations be removed from the weblate side? I wasn't sure if it was the "sentinel" version and is only editable on this end. I don't want to break weblate again :grimacing:

BentiGorlich commented:
I want to wait for @TheVillageGuy and @melroy89 to review the changes before I merge them. It is not that small of a change anymore :D

asdfzdfj commented:
my 2c after test driving for a day for consideration (desktop, fixed width):

- given the amount of associated configuration options I think it should be grouped into a new section, if anything this should give more space for description and not having to include the word 'subscription' in almost all of them.
- highlighting current active magazine is a nice touch, however I find using hard border to mark it is quite jarring (more than once that I mistook this as browser keyboard navigation border that's stuck), I tried modifying it to have it change background color of active magazine instead and I find it less jarring (and ideally using the swap/popout icon color for text highlight too, but css's a mess and I don't know which color is it)
  ![image](https://github.com/MbinOrg/mbin/assets/20770492/93443041-a2cc-4ef6-8bdb-dc9d0a281542)
- the split panel width feels a bit scuffed, from my experiment I think the column width could be safely raised to about `1.25fr` and still looks fine without eating too much into main content's width (see above image)
- in single entry/post view, the inline subscription panel should be at the top, or at least not between thread and magazine info
  ![image](https://github.com/MbinOrg/mbin/assets/20770492/2ea07a03-bba1-499b-9ecd-1fe6bd6c0593)
  (a dirty inspector edit to show what it could looks like)
- this one's a bit of edge case but for the left split panel case, both side toggle and inline/split toggle have a right pointing arrow in its icon, and that could be a bit hard to read at a glance (occasionally I have to stop and think which one will do what I want at the moment), I wonder if there's other icon pair that could be used for the inline/split panel toggle instead of the current one

melroy89 commented:
> I want to wait for @TheVillageGuy and @melroy89 to review the changes before I merge them. It is not that small of a change anymore :D

Thanks. I will come back soon with a list of feedback/ideas/proposals I have for this PR if you don't mind 👍🏽 . Thanks for waiting on me and TheVillageGuy.

TheVillageGuy commented:
> I want to wait for @TheVillageGuy and @melroy89 to review the changes before I merge them. It is not that small of a change anymore :D

Is there somewhere where I can test it? I don't have much time and my Dev server isn't up yet

melroy89 commented:
/me testing on https://kbin.melroy.org/

TheVillageGuy commented:
I just tried it on Melroy's instance and it looks great! The pop/in pop out button could be improved by simply removing it by using just arrows, moving it from left to right. 

So when it''s on the left clicking -> will dock it. When it's docked <- | -> to move it either left or right. When it's on the right <- will dock it

BentiGorlich commented:
> I just tried it on Melroy's instance and it looks great! The pop/in pop out button could be improved by simply removing it by using just arrows, moving it from left to right.
> 
> So when it''s on the left clicking -> will dock it. When it's docked <- | -> to undock it either left or right. When it's on the right <- to dock it

I don't know 🤔 I think its not super intuitive if you don't have to option to just change the side its on when it is already undocked

BentiGorlich commented:
I noticed I never repleid to you @asdfzdfj 
> my 2c after test driving for a day for consideration (desktop, fixed width):
> 
> * given the amount of associated configuration options I think it should be grouped into a new section, if anything this should give more space for description and not having to include the word 'subscription' in almost all of them.

Agreed, I removed some options, but I need to put them back for the mobile view, as the icons are hidden when on mobile

> * highlighting current active magazine is a nice touch, however I find using hard border to mark it is quite jarring (more than once that I mistook this as browser keyboard navigation border that's stuck), I tried modifying it to have it change background color of active magazine instead and I find it less jarring (and ideally using the swap/popout icon color for text highlight too, but css's a mess and I don't know which color is it)

I can definetely understand that. Now I just need a CSS class which achieves what you want, so the themes still look nice

> * the split panel width feels a bit scuffed, from my experiment I think the column width could be safely raised to about `1.25fr` and still looks fine without eating too much into main content's width (see above image)

I don't have an opinion on that. I think it looks fine the way it is, but we can vote on that or smth :D

> * in single entry/post view, the inline subscription panel should be at the top, or at least not between thread and magazine info

Absolutely.
 
> * this one's a bit of edge case but for the left split panel case, both side toggle and inline/split toggle have a right pointing arrow in its icon, and that could be a bit hard to read at a glance (occasionally I have to stop and think which one will do what I want at the moment), I wonder if there's other icon pair that could be used for the inline/split panel toggle instead of the current one

Ok yeah. What do you think about the suggestion from @TheVillageGuy ?

melroy89 commented:
@BentiGorlich one question: Despite the subs being showed by default now. It won't show as a separate panel, right?

melroy89 commented:
@BentiGorlich Another question/request is what if people disabled the icons "Show magazines’ icons" to "False".. Should this sidebar with subs also stop showing the icons on desktop?

melroy89 commented:
Another feedback would be that by default it the sidebar seems like 150px width, but if you have a magazine with a big name it can get extended to 167 px. So my proposal would be: increase the sidebar to this width in all cases ORR.. limit the chars earlier to keep the width the same.

melroy89 commented:
If the sidebar is enabled and put in a separate sidebar, the right sidebar becomes SMALLER. Causing this to happen:

![image](https://github.com/MbinOrg/mbin/assets/628926/ffb4cdf4-cb77-4233-a626-f47a6226a296)

melroy89 reviewed (dismissed):
See my remarks above.

TheVillageGuy commented:
> Last but not least can we remove the LEFT padding on the right panel, and the right padding on the left panel. Making the spacing between the main content and the sidebars the same as the action bar or alerts on the top. Basically now it's double the padding.. ![image](https://user-images.githubusercontent.com/628926/280020953-52b49f75-dc8e-4725-9782-9b87751ca294.png)
> 
> Currently the padding is applied to both left and right on the sidebar classes (eg. `#middle #sidebar`). So only add right padding to the right sidebar (same idea on the left sidebar (`#sidebar .sidebar-subscriptions:not(.inline), .sidebar-subscriptions:not(.inline)` .. why so complex; but I digress?); only add left padding on the left sidebar), and keep `#middle #main` alone (do **not** change the padding on `#middle #main`)
> 
> Ps. so maybe also make it CSS cleaner, why not just add padding to `#sidebar, .sidebar-subscriptions` and remove the padding when inline. Sounds easier for me.

Padding on my instance is asymmetrical, but I assume that's been fixed.

I would like the subscription bar to be the same width as the sidebar. Or at least when it's on the opposite side, for symmetry but also so that longer magazine names will fit

melroy89 commented:
> > Last but not least can we remove the LEFT padding on the right panel, and the right padding on the left panel. Making the spacing between the main content and the sidebars the same as the action bar or alerts on the top. Basically now it's double the padding.. ![image](https://user-images.githubusercontent.com/628926/280020953-52b49f75-dc8e-4725-9782-9b87751ca294.png)
> > Currently the padding is applied to both left and right on the sidebar classes (eg. `#middle #sidebar`). So only add right padding to the right sidebar (same idea on the left sidebar (`#sidebar .sidebar-subscriptions:not(.inline), .sidebar-subscriptions:not(.inline)` .. why so complex; but I digress?); only add left padding on the left sidebar), and keep `#middle #main` alone (do **not** change the padding on `#middle #main`)
> > Ps. so maybe also make it CSS cleaner, why not just add padding to `#sidebar, .sidebar-subscriptions` and remove the padding when inline. Sounds easier for me.
> 
> Padding on my instance is asymmetrical, but I assume that's been fixed.
> 
> I would like the subscription bar to be the same width as the sidebar. Or at least when it's on the opposite side, for symmetry but also so that longer magazine names will fit

the padding is now the same on the left and right side.  But.. my proposal here was specifically to change the sidebar paddings in general. Which would result into this (preview):

![image](https://github.com/MbinOrg/mbin/assets/628926/47c6aa2c-d273-41e2-ab6a-54cafb1f2c94)

TheVillageGuy commented:
> > > Currently the padding is applied to both left and right on the sidebar classes (eg. `#middle #sidebar`). So only add right padding to the right sidebar (same idea on the left sidebar (`#sidebar .sidebar-subscriptions:not(.inline), .sidebar-subscriptions:not(.inline)` .. why so complex; but I digress?); only add left padding on the left sidebar), and keep `#middle #main` alone (do **not** change the padding on `#middle #main`)
> > > Ps. so maybe also make it CSS cleaner, why not just add padding to `#sidebar, .sidebar-subscriptions` and remove the padding when inline. Sounds easier for me.
> > 
> > 
> > Padding on my instance is asymmetrical, but I assume that's been fixed.
> > I would like the subscription bar to be the same width as the sidebar. Or at least when it's on the opposite side, for symmetry but also so that longer magazine names will fit
> 
> the padding is now the same on the left and right side. But.. my proposal here was specifically to change the sidebar paddings in general. Which would result into this (preview), do you like this or not?:
> 
> ![image](https://user-images.githubusercontent.com/628926/280065931-47c6aa2c-d273-41e2-ab6a-54cafb1f2c94.png)
> 
> Ps. maybe it's a bad idea.. I was just sharing my ideas.

Looks good to me

BentiGorlich commented:
> @BentiGorlich one question: Despite the subs being showed by default now. It won't show as a separate panel, right?

correct

BentiGorlich commented:
> @BentiGorlich Another question/request is what if people disabled the icons "Show magazines’ icons" to "False".. Should this sidebar with subs also stop showing the icons on desktop?

No not right now. Should propably be the case.

BentiGorlich commented:
> Another feedback would be that by default it the sidebar seems like 150px width, but if you have a magazine with long names it can get extended to 167 px. So my proposal would be: increase the sidebar to this width in all cases ORR.. limit the chars earlier to keep the width the same.

I thought I fixed that... Need to look at that again

BentiGorlich commented:
> If the sidebar is enabled and put in a separate sidebar, the right sidebar becomes SMALLER. Causing this to happen:
> 
> ![image](https://user-images.githubusercontent.com/628926/280020043-ffb4cdf4-cb77-4233-a626-f47a6226a296.png)

Ok, but what solution do you propose? I personally don't mind the little overflow there :D

BentiGorlich commented:
I'd say general sidebar padding belongs in its own issue :D

melroy89 commented:
I'm using you latest changes now, look this is what I mean:

![fixed_right_sidebar_width_decreases](https://github.com/MbinOrg/mbin/assets/628926/45e73343-9b0d-4460-8fc4-ae43868a4639)

melroy89 commented on templates/layout/sidebar_subscriptions.html.twig:
```diff
@@ -1,30 +1,71 @@
-
-                    {% if magazine.icon %}
-                        <img src="{{ asset(magazine.icon.filePath) | imagine_filter('avatar_thumb') }}" class="magazine-subscription-avatar" alt="{{ magazine.name }}'s icon" />
+<div class="sidebar-subscriptions {{ app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_GENERAL_SHOW_SUBSCRIPTIONS_IN_SEPARATE')) is not same as 'true' ? 'inline' : '' }}">
```
Make this also a `aside` element instead of `div` just like the other sidebar. So it's easier for browsers /read mode to know it's related to the main content, but aside of the main.

melroy89 replied:
ow I see.. that is not possible or is it? Because you reuse the same layout for inside the aside :\

melroy89 replied:
I was even considering make the left sidebar more generic, so we can use it not only for subscriptions but other things...

BentiGorlich replied:
well firefox doesn't complain about an `aside` in an `aside` :D

melroy89 commented:
I don't agree. but I don't want to be blocking. I remove my blocked for changes.

BentiGorlich commented:
> I don't agree. but I don't want to be blocking. I remove my blocked for changes.

What is bugging you? I am not particularly fond of the small sidebar, but the option to make it larger always exist. There already is a setting that could do that when popped out

melroy89 commented:
> I am not particularly fond of the small sidebar,

OK let's make in symmetrical again, but I agree in that case that both sidebars are allowed to become smaller in width, to still give enough space in the main section.

melroy89 commented on assets/styles/layout/_layout.scss:
```diff
@@ -36,7 +36,7 @@ body {
       .subs-show & {
         grid-template-areas: 'subs main sidebar';
-        grid-template-columns: 1fr 3fr 1fr;
+        grid-template-columns: 1fr 6fr 2fr;
```
So I think that will become this again:

```suggestion
       grid-template-columns: 1fr 3fr 1fr;
```

Or we could even for go: `300px auto 300px;`

asdfzdfj commented:
> > my 2c after test driving for a day for consideration (desktop, fixed width):
> > 
> > * given the amount of associated configuration options I think it should be grouped into a new section, if anything this should give more space for description and not having to include the word 'subscription' in almost all of them.
> 
> Agreed, I removed some options, but I need to put them back for the mobile view, as the icons are hidden when on mobile

that'd be great, also I think it would be nice to have all the options available when the subscription panel is enabled instead of showing/hiding options depending on which settings is enabled like it is now, unless you have a need to enforce possible setting state and its transition, which is understandable

(this one is _mainly my personal opinion_ due to a bit of my browsing quirk: I have them set to clear all data on closing, and so when I come back again I have to re-set all options and while the panel button is nice and interactive, it immediately reloads which means I have to set one option at a time, and I'd like to be able to set all needed options and only reload once to apply)
 
---

> Seems super intuitive to me, the arrows indicate in which direction the panel will move. Especially if you make the texts clear, so from left to right
> -> dock in sidepanel
> 
> <- undock to the left
> -> undock to the right
> 
> <- dock in sidepanel
> 
> The next step is to just be able to slide the panels around :)

> > * this one's a bit of edge case but for the left split panel case, both side toggle and inline/split toggle have a right pointing arrow in its icon, and that could be a bit hard to read at a glance (occasionally I have to stop and think which one will do what I want at the moment), I wonder if there's other icon pair that could be used for the inline/split panel toggle instead of the current one
> 
> Ok yeah. What do you think about the suggestion from @TheVillageGuy ?

It's a bit hard for me to imagine how will it work or looks like but it might work out, I think I'll need either working prototype or at least some illustrative figures to give a better assessment

at the very least I was thinking of just changing the panel pop out (inline -> split) button to something like [arrow-up-right-from-square](https://fontawesome.com/icons/arrow-up-right-from-square?f=classic&s=solid) and the reattach (split -> inline) to what should be the inverse of the pop out button, but I couldn't find one and the next best thing I could find is [down-left-and-up-right-to-center](https://fontawesome.com/icons/down-left-and-up-right-to-center?f=classic&s=solid) and for me this is sufficient.

![image](https://github.com/MbinOrg/mbin/assets/20770492/bd065e29-4c7e-494b-8a0e-5271e2980517)
![image](https://github.com/MbinOrg/mbin/assets/20770492/6b11ebf8-d795-465b-a71a-1e507fa43735)
(dirty inspector edit again to show what it might look like)

---

> > If the sidebar is enabled and put in a separate sidebar, the right sidebar becomes SMALLER. Causing this to happen:
> > ![image](https://user-images.githubusercontent.com/628926/280020043-ffb4cdf4-cb77-4233-a626-f47a6226a296.png)
> 
> Ok, but what solution do you propose? I personally don't mind the little overflow there :D

you could try to play around with [`minmax()`](https://developer.mozilla.org/en-US/docs/Web/CSS/minmax) in grid column size definition, maybe something like `grid-template-columns: 1fr 6fr minmax(320px, 2fr);` and see how it works out

TheVillageGuy commented:
> > I am not particularly fond of the small sidebar,
> 
> OK let's make in symmetrical again, but I agree in that case that both sidebars are allowed to become a smaller in width, to still give enough space in the main section. What do you think? You agree as well?
> 
> EDIT: Sorry that I changed my mind, you convinced me ;)

Why, it works and looks fine as is, why should a website be narrow?

asdfzdfj commented on templates/base.html.twig:
```diff
@@ -24,6 +24,31 @@
 
     {% block stylesheets %}
         {{ encore_entry_link_tags('app') }}
+            <style>
+                {#
+                This is needed so the magazine names in the subscription panel are correctly overflowing.
+                The max width depends on the max-width of the whole container, which is controlled by a setting (KBIN_PAGE_WIDTH)
+                 #}
+                .subscription-list .subscription a {
+                    {% if app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_SUBSCRIPTIONS_IN_SEPARATE_SIDEBAR')) is same as 'true' %}
+                        {% if app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_PAGE_WIDTH')) is same as constant('App\\Controller\\User\\ThemeSettingsController::MAX') %}
+                            max-width: calc(100vw / 9 - 3rem);
+                        {% elseif app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_PAGE_WIDTH')) is same as constant('App\\Controller\\User\\ThemeSettingsController::AUTO') %}
+                            max-width: calc(85vw / 9 - 3rem);
+                        {% else %}
+                            max-width: calc(1360px / 9 - 3rem);
+                        {% endif%}
+                    {% else %}
+                        {% if app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_PAGE_WIDTH')) is same as constant('App\\Controller\\User\\ThemeSettingsController::MAX') %}
+                            max-width: calc(100vw / 4 - 3.5rem);
+                        {% elseif app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_PAGE_WIDTH')) is same as constant('App\\Controller\\User\\ThemeSettingsController::AUTO') %}
+                            max-width: calc(85vw / 4 - 3.5rem);
+                        {% else %}
+                            max-width: calc(1360px / 4 - 3.5rem);
+                        {% endif%}
+                    {% endif %}
+                }
+            </style>
```
highly recommended to find a way to put this rule into the stylesheet

from what I could see, setting page width to either auto or max would set `.width--{auto,max}` on the parent container of the subscription sidebar, and when subscription sidebar is inline it'll have `.inline` class set, so I think it should be possible to define the default values for fixed width container and then add rules after to make setting specific override:

```scss
.subscription-list {
  /* defaults for fixed width container */
}

// setting specific override rules
.subscription-list {
  .width--auto & {
    // auto container panel override
    &.inline {
      // auto inline panel override
    }
  }
  .width--max & {
    // max container panel override
    &.inline {
      // max inline panel override
    }
  }
}
```

BentiGorlich replied:
I'll look into that. That would be a far better solution, I agree

melroy89 commented:
> Why make the bars smaller, it works and looks fine as is, why should a website be narrow?

No, the **main** content should be bigger, _not_ narrower, that was my point when two sidebars are active.

BentiGorlich commented:
> at the very least I was thinking of just changing the panel pop out (inline -> split) button to something like [arrow-up-right-from-square](https://fontawesome.com/icons/arrow-up-right-from-square?f=classic&s=solid) and the reattach (split -> inline) to what should be the inverse of the pop out button, but I couldn't find one and the next best thing I could find is [down-left-and-up-right-to-center](https://fontawesome.com/icons/down-left-and-up-right-to-center?f=classic&s=solid) and for me this is sufficient.
> 
> ![image](https://user-images.githubusercontent.com/20770492/280203065-bd065e29-4c7e-494b-8a0e-5271e2980517.png) ![image](https://user-images.githubusercontent.com/20770492/280202832-6b11ebf8-d795-465b-a71a-1e507fa43735.png) (dirty inspector edit again to show what it might look like)

@TheVillageGuy @melroy89 @e-five256 do you agree that these icons would be better suited? I am not really sure. I like both the way it is right now and @asdfzdfj suggestion

BentiGorlich commented:
I am playing around with the width right now and I gotta say I am tempted to just hardcode the width of both sidebars to 305px. The "active user" panel has a min width of 302.xx px anyway. I don't know if it is all that useful to have the sidebars grow with the middle section when one chooses a wider page width...

e-five256 commented:
> I am playing around with the width right now and I gotta say I am tempted to just hardcode the width of both sidebars to 305px. The "active user" panel has a min width of 302.xx px anyway. I don't know if it is all that useful to have the sidebars grow with the middle section when one chooses a wider page width...

That seems fine to me. When I added the page width thing, I did notice the sidebar grew in width as well, but I didn't have a strong opinion about that as I thought maybe it could add some breathing room; but the main goal of page width to me was to allow a larger main content box so text wasn't so squished (especially when you get into super nested comment chains). The sidebar width staying consistent in each page width setting seems fine

e-five256 commented:
> do you agree that these icons would be better suited?

to me, the pop out icon is usually very specific to opening something in a separate window; so if I clicked that I would expect the subscription panel to open in a small new browser window that was separate but clicking on it could target the main browser or something. I feel like what's happening here is more of a "split" than a "pop out", but that's just my opinion

BentiGorlich commented:
> > do you agree that these icons would be better suited?
> 
> to me, the pop out icon is usually very specific to opening something in a separate window; so if I clicked that I would expect the subscription panel to open in a small new browser window that was separate but clicking on it could target the main browser or something. I feel like what's happening here is more of a "split" than a "pop out", but that's just my opinion

Makes sense to me. I'll implement the button system @TheVillageGuy suggested and maybe the confusion will gone afterwards :D

nobodyatroot commented on translations/messages.en.yaml:
```diff
@@ -790,10 +790,15 @@ password_and_2fa: Password & 2fa
+subscription_sidebar_pop_out_left: Move to separate sidebar on the left
+subscription_sidebar_pop_in: Move subscriptions to the inline panel
+subscription_panel_large: Large panel
+subscription_header: Subs
```
Since we have the space (and to be consistent with the "random magazines" header), should we rename this to "subscribed magazines"?

BentiGorlich replied:
Yeah since I have changed the sizes again this doesn't make any sense any more

e-five256 commented on templates/layout/sidebar_subscriptions.html.twig:
```diff
@@ -1,30 +1,78 @@
-        {% for magazine in magazines %}
-            <div class="subscription {{ openMagazine and openMagazine.name is same as magazine.name ? 'active' : '' }}">
-                <a href="/m/{{ magazine.name }}">
+<aside class="sidebar-subscriptions {{ app.request.cookies.get(constant('App\\Controller\\User\\ThemeSettingsController::KBIN_SUBSCRIPTIONS_IN_SEPARATE_SIDEBAR')) is not same as 'true' ? 'inline' : '' }}">
```
small nitpick, could we add a `{% if magazines|length %}` and `{% endif %}` at the end? I _think_ that should exclude this template if the user has no subscriptions. not a big deal but seems like it can be hidden until they sub to something

![image](https://github.com/MbinOrg/mbin/assets/146029455/de1cb2b5-4228-4d4a-b2f0-21c44a340de8)

e-five256 replied:
actually that might be confusing with the options in the sidebar still... maybe we can hold off on that for another PR pass, this one has already gotten quite a lot in it

BentiGorlich replied:
I'd put it in a new PR as well. But I'd prefer it if there was just a text inside: "you did not yet subscribe to any magazine".
